poetry/packages/locker.py

```diff
@@ -561,12 +561,11 @@ def _dump_package(self, package):  # type: (Package) -> dict
         if package.source_url:
             url = package.source_url
             if package.source_type in ["file", "directory"]:
-                # The lock file should only store paths relative to the root project
-                url = Path(
-                    os.path.relpath(
-                        Path(url).as_posix(), self._lock.path.parent.as_posix()
-                    )
-                ).as_posix()
+                # For all local paths, the lock file should store the relative path from the root folder
+                url = Path(url).as_posix()
+                package_root = self._lock.path.parent
+                if url.startswith(".") or Path(url).anchor == package_root.anchor:
```

Comment:
Should this be using an "is absolute" check instead? Additionally, I am not sure if this is the right fix here. Am I correct in assuming that the intention here is to not modify the path if is a network drive?

Reply:
Hi @abn, thanks for reviewing

My intention was to preserve the original logic of storing the relative path when possible. This would apply only to paths that were relative already or had the same root. This passes absolute paths not on the local file system unmodified

Using `.is_absolute()` would be true for both a `C:\Path` and a `Z:\Path` but the latter cannot be a relative path because it is on a different drive. In the latter case, the path should properly be configured by the user and passed through

Let me know if there are other test cases to add or you would like to see a different approach

I think there could be an argument to treat the toml file as the source of truth for file paths where the locker should use the user's path without modification. Even if the shortest relative path is calculated for the lock file, the toml would still have a specific path that should have been relative

Alternatively, this could be implemented with a try/except rather than a "look before you leap" approach

Reply:
@abn / @finswimmer anything I can do to help move this change along?

Reply:
Apologies for not getting to this. Will get to it later today.